ownCloud/Viewer/OCExtension Subclasses/OCDisplayExtension.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  OCDisplayExtension.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 03/09/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+
+class OCDisplayExtension: OCExtension {
+
+	init(identifier: OCExtensionIdentifier!, type: OCExtensionType!, location: OCExtensionLocation!, features: [String : Any]!, objectProvider: @escaping OCExtensionObjectProvider) {
+
+		super.init()
+		self.identifier = identifier
+		self.type = type
+		self.locations = [location]
+		self.features = features
+		self.objectProvider = objectProvider
+	}
+
+	override func matchesContext(_ context: OCExtensionContext!) -> OCExtensionPriority {
+
+		var matchPriority: OCExtensionPriority = .noMatch
+
+		if context.location.type == self.type {
```

Comment:
@pablocarmu For me the below code is not self-explaining without comments

Reply:
it's a Swift translation of [this](https://github.com/owncloud/ios-sdk/blob/feature/filemgmt/ownCloudSDK/Extensions/OCExtension.m).

It does the same but also in the location comparison it compares the mime-type of the file you want to present and if it matches, the context match and it's returned as a possible matched value.

it goes through the location, the features and the priority.

Reply:
If you put the MIME-Type into the locationIdentifier, there's no longer any need to override `matchesContext`.

As far as I can see, the locationIdentifier is not really in use for anything right now, so that's what I'd propose.